include/boost/url/detail/params_table.hpp

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+// Copyright (c) 2021 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_DETAIL_PARAMS_TABLE_HPP
+#define BOOST_URL_DETAIL_PARAMS_TABLE_HPP
+
+#include <boost/url/string_view.hpp>
+#include <boost/assert.hpp>
+#include <cstdint>
+#include <type_traits>
+
+namespace boost {
+namespace urls {
+
+namespace detail {
+
+//------------------------------------------------
+
+// params array stored at the
+// end of allocated url string data
+
+struct params_table_entry
+{
+    off_t kp;   // key pos
+    off_t kn;   // key size
+    off_t vp;   // value pos
```

Comment:
well if you know the key size then you also know the value pos, right? and if you know the position of the next key then that gives you the value length?